zerver/tests/test_decorators.py

```diff
@@ -759,6 +760,35 @@ def test_login_deactivated_user(self):
             "Sorry for the trouble, but your account has been deactivated",
             result)
 
+    def test_login_deactivated_mirror_dummy(self):
+        # type: () -> None
+        """
+        logging in fails with an inactive user
+
+        """
+        user_profile = self.example_user('hamlet')
+        user_profile.is_mirror_dummy = True
+        user_profile.save()
+
+        password = initial_password(user_profile.email)
+        request = mock.MagicMock()
+        request.get_host.return_value = 'testserver'
+        form = OurAuthenticationForm(request,
+                                     data={'username': user_profile.email,
+                                           'password': password})
+        with self.settings(AUTHENTICATION_BACKENDS=('zproject.backends.EmailAuthBackend',)):
+            self.assertTrue(form.is_valid())
+
+        do_deactivate_user(user_profile)
+        user_profile.save()
+
+        form = OurAuthenticationForm(request,
+                                     data={'username': user_profile.email,
+                                           'password': password})
+        with self.settings(AUTHENTICATION_BACKENDS=('zproject.backends.EmailAuthBackend',)):
+            self.assertFalse(form.is_valid())
+            self.assertIn("Please enter a correct email", str(form.errors))
+
```

Comment:
Let's add a final assert that if `is_mirror_dummy` is now set back to `False`, now the form isn't valid with that email.